src/node/db/AuthorManager.js

```diff
@@ -107,6 +107,10 @@ exports.doesAuthorExists = exports.doesAuthorExist;
  * @param {String} token The token
  */
 exports.getAuthor4Token = async (token) => {
+  // e.g. in case of an export of a public pad, a token is not needed
+  if (token == null) {
```

Comment:
Wouldn't it be better to change the caller to not call this function if `token` is `null`?

Reply:
I thought about this but choosed to do it here, because there are many callers of that function in plugins. but now, that I write this line, I think I need to check the (plugin) callers anyway, because until now `getAuthor4Token` never returned null/undefined, so plugins could break when they don't expect null/undefined

Reply:
I checked all plugins that are available via static.etherpad.org:
The following seem to be save (explicitly checking for token before calling `getAuthor4Token`):
ep_auth_author
ep_codepad
ep_headerauth
ep_httpauth_author
ep_idea_review
ep_jwt
ep_ldapauth
ep_pomerium
ep_termin_app_integration

The following are save because they explicitly test the message.type to be `CLIENT_READY` (and PadMessageHandler checks for token in the `CLIENT_READY` case):
ep_openid_connect
ep_readonly_guest

The following could be vulnerable, need to check again tomorrow:
ep_cloudron and ep_hash_auth. They seem to register the `handleMessage` hook, while not checking if they parse an `CLIENT_READY` message. Other messages like `USER_CHANGES` don't have a token in `context.message`. I wonder if this is problematic ie it looks to me as if they get the same "undefined" author (which is a real authorId) for any requests...

Reply:
Thinking about it more, I think the best approach would be to throw if given a falsy value (after updating plugins as necessary):
```javascript
  if (!token) throw new Error('missing token');
```